### ISSUE TITLE
External Reference Metadata for cruise.json and file.json

### ISSUE THREAD

**Issue by DocOtak** [cchdo/cchdo.ucsd.edu#26]
### Overview

A method is needed to track the external references to our cruise and file objects.
This proposal supersedes issue #25.

Here is a blank example:

``` javascript
{
  "external_refs": [
    {
      "organization": "",
      "type": "",
      "value": "",
      "properties": {} // optional properties object key (like geojson).
    }
  ]
// other keys needed for a valid cruise.json or file.json
}
```

Note that it is a list of these "reference objects". 
The `organization` key should have all lowercase letters in it (e.g. "nsf" rather than "NSF").
The `type` key should have one of the following values:
- `"doi"` if the value should be interpreted as a DOI (document object identifier)
- `"ark"` if the value should be interpreted as an ARK (archive resource key)
- `"alias"` if the value is some external name for the same object, note that an external name can refer to multiple instances of our objects and multiple external aliases can refer to a single one of our objects.
- `"award"` if the value is the "award" or "grant" number from some funding agency.
- `"accession"` if the NECI accession number, this one is special because it designates that something has been stored in the national archive.
### Rationale

This structure is based on the ability to utilize the `@>` and `<@` operators in postgresql.
A query for some identifier would be quick and easy, for example: 
`SELECT cruise WHERE cruise.json @> '{"external_refs"[{"organization":"nsf"}]'::jsonb` 
would find all the cruise objects which contain "nsf" as the organization. Notice that it is only needs part of the object to perform the query.

It also relies on the idea of "award numbers" just being some external organizations identifier for objects we are tracking. That is, it may not be unreasonable to consider all the cruises done as part of some NSF grant to all be identified by the NSF award.
#### NSF grant

``` javascript
{
  "external_refs": [
    {
      "organization": "nsf",
      "type": "award",
      "value": "1234567890",
      "properties": {"dtstart":"2010-01", "dtend":"2015-01"}
    }
  ]
// other keys needed for a valid cruise.json or file.json
}
```
#### NCEI (formerly NODC)

``` javascript
{
  "external_refs": [
    {
      "organization": "ncei",
      "type": "accession",
      "value": "1234567890",
      "properties": {"date":"2015-07-01"}
    }
  ]
// other keys needed for a valid cruise.json or file.json
}
```
#### EZID (part of the California Digital Library)

``` javascript
{
  "external_refs": [
    {
      "organization": "EZID",
      "type": "doi",
      "value": "10.7289/V5154F01"
    }
  ]
// other keys needed for a valid cruise.json or file.json
}
```
#### EZID (part of the California Digital Library)

``` javascript
{
  "external_refs": [
    {
      "organization": "EZID",
      "type": "ark",
      "value": "http://myrepo.example.org/ark:/12345/bcd987"
    }
  ]
// other keys needed for a valid cruise.json or file.json
}
```
#### BCO-DMO

``` javascript
{
  "external_refs": [
    {
      "organization": "bco-dmo",
      "type": "alias",
      "value": "MV1305"
    }
  ]
// other keys needed for a valid cruise.json or file.json
}
```
#### R2R

``` javascript
{
  "external_refs": [
    {
      "organization": "r2r",
      "type": "alias",
      "value": "MV1305"
    }
  ]
// other keys needed for a valid cruise.json or file.json
}
```

Notice that two or more organizations may have the same identifier for some object.
#### CDIAC

``` javascript
{
  "external_refs": [
    {
      "organization": "cdiac",
      "type": "alias",
      "value": "318M20130320"
    }
  ]
// other keys needed for a valid cruise.json or file.json
}
```

Notice that the CDIAC alias might be exactly the same as a CCHDO expocode.

**Comment by DocOtak** [comment]
Perhaps the key should be called simply `"references"`

**Comment by DocOtak** [comment]
Additionally, `citations` should also be a reference type.

Example:

``` javascript
{
  "external_refs": [
    {
      "organization": "",
      "type": "citation",
      "value": "Sagarin, B. J., & Lawler-Sagarin, K. A. (2005). Critically evaluating competing theories: An exercise based on the Kitty Genovese murder. Teaching of Psychology, 32(3), 167–169. http://dx.doi.org/10.1207/s15328023top3203_8",
      "properties": {
        "bibtex": "@article{}"
      }
    }
  ]
// other keys needed for a valid cruise.json or file.json
}
```

**Comment by githubbuser** [comment]
Would be helpful if the documentation ( hosted online?) is kept up-to-date as to all possible "organization" and "type" values.  Similar to the parameters.json file  documenting every parameter that is known (hosted) by CCHDO.   Otherwise,   great start!

**Comment by DocOtak** [comment]
Indeed, the docs will be kept up to date with regards to what the cruise.json and file.json objects can look like.

**Comment by DocOtak** [comment]
Resolved by 08d57b7, the docs have been updated as well.
